packages/sanity/src/core/store/events/getDocumentEvents.ts

```diff
@@ -0,0 +1,320 @@
+}
+
+// This assumes the view is from the publishedDocument having only drafts. (Versions are not yet supported here)
+function getEventFromTransaction(
+  documentId: string,
+  transaction: Transaction,
+  previousTransactions: Transaction[],
+): DocumentGroupEvent | null {
+  const base = {
+    timestamp: transaction.timestamp,
+    author: transaction.author,
+  }
+
+  const draftId = getDraftId(documentId)
+  const publishedId = getPublishedId(documentId)
+  const draftEffect = transaction.effects[draftId]
+  const publishedEffect = transaction.effects[publishedId]
+  const draftState = getEffectState(draftEffect)
+  const publishedState = getEffectState(publishedEffect)
+
+  const {type, document} = STATE_MAP[publishedState][draftState]
+
+  switch (type) {
+    case 'document.createVersion': {
+      return {
+        ...base,
+        type,
+
+        documentId: documentId,
+        versionId: document === 'draft' ? draftId : publishedId,
+        releaseId: getVersionFromId(documentId),
+        versionRevisionId: transaction.id,
+      }
+    }
+    case 'document.editVersion': {
+      return {
+        ...base,
+        type,
+
+        releaseId: getVersionFromId(documentId),
+        versionId: draftId,
+        versionRevisionId: transaction.id,
+      }
+    }
+
+    case 'document.deleteVersion': {
+      return {
+        ...base,
+        type,
+
+        versionId: document === 'draft' ? draftId : publishedId,
+        // TODO: Should this be the actual transaction id that created the draft document?
+        // The revision id of the last edit in the draft document
+        versionRevisionId:
+          previousTransactions.find((t) => t.documentIDs.includes(draftId))?.id || 'not-found',
+        releaseId: getVersionFromId(documentId),
+      }
+    }
+
+    case 'document.publishVersion': {
+      return {
+        ...base,
+        type,
+
+        revisionId: transaction.id,
+        releaseId: getVersionFromId(documentId),
+        versionId: draftId, // TODO: How to get the version in case of releases
+        versionRevisionId:
+          previousTransactions.find((t) => t.documentIDs.includes(draftId))?.id || 'not-found',
+        cause: {
+          type: 'document.publish', // TODO: How to get the `release.publish` and the `release.schedule` events?
+        },
+      }
+    }
+
+    case 'document.unpublish': {
+      return {
+        ...base,
+        type,
+
+        // TODO: How to identify if we are creating a version by the unpublish, is that even possible?
+        // The version that will be created by this unpublish action, e.g. drafts.foo
+        versionId: transaction.documentIDs.length > 1 ? draftId : undefined,
+        // TODO: Should this be the actual transaction id that created the draft document?
+        // The revision id of the last edit in the publish document
+        versionRevisionId:
+          previousTransactions.find((t) => t.documentIDs.includes(publishedId))?.id || 'not-found',
```

Comment:
use here `transaction.id`? 

It's not clear for me if we want to have here the id of the transaction that last touched the published document or the transaction id when this version document was created.